logging/troubleshooting/cluster-logging-troubleshooting-for-critical-alerts.adoc

```diff
@@ -0,0 +1,536 @@
+
+
+[id="elasticsearch-node-disk-flood-watermark-reached"]
+== Elasticsearch Node Disk Flood Watermark Reached
+
+Elasticsearch enforces a read-only index block on every index that has both of these conditions:
+
+* One or more shards are allocated to the node.
+* One or more disks exceed the https://www.elastic.co/guide/en/elasticsearch/reference/6.8/disk-allocator.html[flood stage].
+
+.Troubleshooting
+
+. Check the disk space of the Elasticsearch node.
++
+[source,terminal]
+----
+for pod in `oc -n openshift-logging get po -l component=elasticsearch -o jsonpath='{.items[*].metadata.name}'`; do echo $pod; oc -n openshift-logging exec -c elasticsearch $pod -- df -h /elasticsearch/persistent; done
+----
++
+Check the `nodes.node_name.fs` field to determine the free disk space on that node.
+
+. If the used disk percentage is above 95%, it signifies that the node has crossed the flood watermark. Writing is blocked for shards allocated on this particular node.
+
+. Try to increase the disk space on all nodes.
+
+. If increasing the disk space is not possible, try adding a new data node to the cluster.
+
+. If adding a new data node is problematic, decrease the total cluster redundancy policy.
+
+.. Check the current `redundancyPolicy`.
++
+[source,terminal]
+----
+oc -n openshift-logging get es elasticsearch -o jsonpath='{.spec.redundancyPolicy}'
+----
++
+NOTE: If you are using a `ClusterLogging` CR, enter:
++
+[source,terminal]
+----
+oc -n openshift-logging get cl -o jsonpath='{.items[*].spec.logStore.elasticsearch.redundancyPolicy}'
+----
+
+.. If the cluster `redundancyPolicy` is higher than `SingleRedundancy`, set it to `SingleRedundancy` and save this change.
+
+. If the preceding steps do not fix the issue, delete the old indices.
+
+.. Check the status of all indices on Elasticsearch.
++
+[source,terminal]
+----
+oc exec -n openshift-logging -c elasticsearch <elasticsearch_pod_name> -- indices
+----
+
+.. Identify an old index that can be deleted.
+
+.. Delete the index.
++
+[source,terminal]
+----
+oc exec -n openshift-logging -c elasticsearch <elasticsearch_pod_name> -- es_util --query=<elasticsearch_index_name> -X DELETE
+----
+
+ . Continue freeing up and monitoring the disk space until the used disk space drops below 90%. Then, unblock write to this particular node.
++
+[source,terminal]
+----
+oc exec -n openshift-logging -c elasticsearch <elasticsearch_pod_name> -- es_util --query=_all/_settings?pretty -X PUT -d '{"index.blocks.read_only_allow_delete": null}'
+----
+
+.Additional resources
+
+* Search for "redundancyPolicy" in the "Sample `ClusterLogging` custom resource (CR)" in xref:../../logging/config/cluster-logging-configuring-cr.adoc#cluster-logging-configuring-crd_cluster-logging-configuring-cr[About the Cluster Logging custom resource]
```

Comment:
Same issue as above. Your link works, but it takes me to a page which gives me a dead link.